src/core/MOM_PressureForce_FV.F90

```diff
@@ -1173,19 +1179,22 @@ subroutine PressureForce_FV_Bouss(h, tv, PFu, PFv, G, GV, US, CS, ALE_CSp, p_atm
                     G%HI, GV, tv%eqn_of_state, US, CS%use_stanley_pgf, dpa(:,:,k), intz_dpa(:,:,k), &
                     intx_dpa(:,:,k), inty_dpa(:,:,k), &
                     MassWghtInterp=CS%MassWghtInterp, &
-                    use_inaccurate_form=CS%use_inaccurate_pgf_rho_anom, Z_0p=Z_0p)
+                    use_inaccurate_form=CS%use_inaccurate_pgf_rho_anom, Z_0p=Z_0p, &
+                    MassWghtInterpVanOnly=CS%MassWghtInterpVanOnly, h_nv=CS%h_nonvanished)
```

Comment:
This line and the other `call int_density_dz_...()` lines are dimensionally inconsistent.  `h_nv` takes arguments with units of `[Z ~> m]`, but `CS%h_nonvanished`  is in units of `[H ~> m or kg m-2]`.  This could be corrected by setting and using another temporary variable such as `dz_nonvanished = GV%H_to_Z*CS%h_nonvanished`.